File: app/src/main/java/com/letianpai/robot/desktop/callback/ModeChangeCmdCallback.java
package com.letianpai.robot.desktop.callback;


/**
 * Created by liujunbin
 */

public class ModeChangeCmdCallback {

    private ModeChangeCommandListener mModeChangeCommandListener;

    private static class ModeChangeCmdCallbackHolder {
        private static ModeChangeCmdCallback instance = new ModeChangeCmdCallback();
    }

    private ModeChangeCmdCallback() {
    }

    public static ModeChangeCmdCallback getInstance() {
        return ModeChangeCmdCallbackHolder.instance;
    }

    public interface ModeChangeCommandListener {
        void onRobotModeChange(String command, String data);
    }

    public void setModeChangeCommandListener(ModeChangeCommandListener listener) {
        this.mModeChangeCommandListener = listener;
    }

    public void changeRobotMode(String command, String data) {
        if (mModeChangeCommandListener != null) {
            mModeChangeCommandListener.onRobotModeChange(command, data);
        }
    }

}

File: app/src/main/java/com/letianpai/robot/desktop/callback/ModeChangeCmdCallback.kt
package com.letianpai.robot.desktop.callback


/**
 * Created by liujunbin
 */
class ModeChangeCmdCallback private constructor() {
    private var mModeChangeCommandListener: ModeChangeCommandListener? = null

    private object ModeChangeCmdCallbackHolder {
        val instance: ModeChangeCmdCallback = ModeChangeCmdCallback()
    }

    fun interface ModeChangeCommandListener {
        fun onRobotModeChange(command: String, data: String)
    }

    fun setModeChangeCommandListener(listener: ModeChangeCommandListener?) {
        this.mModeChangeCommandListener = listener
    }

    fun changeRobotMode(command: String, data: String) {
        if (mModeChangeCommandListener != null) {
            mModeChangeCommandListener!!.onRobotModeChange(command, data)
        }
    }

    companion object {
        val instance: ModeChangeCmdCallback
            get() = ModeChangeCmdCallbackHolder.instance
    }
}












